### PR TITLE
Testsets for test/socket

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by KristofferC** [comment]
```
  Got an exception of type UndefVarError outside of a @test
  UndefVarError: d not defined
```

File: test/socket.jl [hunk 1]
```diff
@@ -232,101 +245,105 @@ let
...
-    a = UDPSocket()
-    b = UDPSocket()
-    bind(a, ip"::1", UInt16(randport))
-    bind(b, ip"::1", UInt16(randport + 1))
-
-    tsk = @async begin
-        @test begin
-            (addr, data) = recvfrom(a)
-            addr == ip"::1" && String(data) == "Hello World"
+ d
```
Comment (fredrikekre) [inline test/socket.jl]: Here is the undefined `d` :laughing:

**Review by fredrikekre — APPROVED** [review]

**Comment by fredrikekre** [comment]
#23796

File: test/socket.jl [hunk 2]
```diff
@@ -1,178 +1,190 @@
...
-@test_throws InexactError Int16(IPv4("1.2.3.4"))
-@test_throws InexactError Int64(IPv6("2001:1::2"))
-
-let ipv = parse(IPAddr, "127.0.0.1")
-    @test isa(ipv, IPv4)
-    @test ipv == ip"127.0.0.1"
-end
+@testset "parsing" begin
+    @test ip"127.0.0.1" == IPv4(127,0,0,1)
+    @test ip"192.0" == IPv4(192,0,0,0)
```
Comment (solson) [inline test/socket.jl]: Is there a reference for this behavior? I found it kind of surprising and I'm investigating a few related IPv4 parsing oddities right now. My expectation would be to throw `ArgumentError` for this case and any other with less than 4 octets.

Comment (solson) [reply test/socket.jl]: Just did a bit more digging, and it looks like this is intentionally matching traditional `inet_aton` behavior, which allows the final number to be "the rest of the bits". E.g.

```julia
julia> map(Int, [0x01020304, 0x020304, 0x0304, 0x04])
4-element Array{Int64,1}:
 16909060
   131844
      772
        4

julia> [ip"16909060", ip"1.131844", ip"1.2.772", ip"1.2.3.4"]
4-element Array{Sockets.IPv4,1}:
 ip"1.2.3.4"
 ip"1.2.3.4"
 ip"1.2.3.4"
 ip"1.2.3.4"
```

So I guess there's no issue.

**Pull request merged**